schema/denm/denm_schema_2-1-0.json

```diff
@@ -0,0 +1,579 @@
+      ],
+      "properties": {
+        "protocol_version": {
+          "type": "integer",
+          "description": "version of the ITS message and/or communication protocol",
+          "minimum": 0,
+          "maximum": 255,
+          "examples": [
+            1
+          ]
+        },
+        "station_id": {
+          "type": "integer",
+          "description": "identifier for an ITS-S",
+          "minimum": 0,
+          "maximum": 4294967295,
+          "examples": [
+            1,
+            42
+          ]
+        },
+        "management_container": {
+          "type": "object",
+          "required": [
+            "action_id",
+            "detection_time",
+            "reference_time",
+            "event_position"
+          ],
+          "properties": {
+            "action_id": {
+              "type": "object",
+              "required": [
+                "originating_station_id",
+                "sequence_number"
+              ],
+              "properties": {
+                "originating_station_id": {
+                  "type": "integer",
+                  "description": "identifier of an its station",
+                  "minimum": 0,
+                  "maximum": 4294967295
+                },
+                "sequence_number": {
+                  "type": "integer",
+                  "description": "The sequence number is set each time a new DENM is created. It is used to differentiate from events detected by the same ITS-S.",
+                  "minimum": 0,
+                  "maximum": 65535
+                }
+              }
+            },
+            "detection_time": {
+              "type": "integer",
+              "description": " Time at which the event is detected by the originating ITS-S. For the DENM repetition, this DE shall remain unchanged.",
+              "minimum": 0,
+              "maximum": 4398046511103,
+              "$comment": "Remains the same on repetition; updated if any value changed in the payload, meaning sequence_number remains unchanged but anything else has been updated",
+              "$comment": "Unit: millisecond since ETSI epoch (2004/01/01, so 1072915200000); utcStartOf2004(0), oneMillisecAfterUTCStartOf2004(1)"
+            },
+            "reference_time": {
+              "type": "integer",
+              "description": "Time at which a new DENM, an update DENM or a cancellation DENM is generated.",
+              "minimum": 0,
+              "maximum": 4398046511103,
+              "$comment": "Unit: millisecond since ETSI epoch (2004/01/01, so 1072915200000). utcStartOf2004(0), oneMillisecAfterUTCStartOf2004(1)",
+              "$comment": "Different for each message, even for repetition"
+            },
+            "termination": {
+              "type": "integer",
+              "description": "isCancellation(0), isNegation (1)",
+              "minimum": 0,
+              "maximum": 1
+            },
+            "event_position": {
```

Comment:
Maybe we should create an ETSI_CDD (common data dictionary) JSON schema that would contain a common position definition for CAM, DENM, CPM.

Reply:
Yes I thought about it, but I don't know it these are part of it
DENM repository has its own CDD where they come from

We first need to list what object are common to all the schemas

Reply:
Indeed, let's keep this for future 3.+ versions...